packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-self-reg/java/integrationsteps.md

```diff
@@ -1,3 +1,251 @@
+   modelAndView.addObject("canSkip", canSkip);
+   List<String> factorMethods = new ArrayList<>();
+   for (Authenticator authenticator : response.getAuthenticators()) {
+      for (Authenticator.Factor factor : authenticator.getFactors()) {
+            factorMethods.add(factor.getMethod());
+      }
+   }
+   session.setAttribute("authenticators", response.getAuthenticators());
+   modelAndView.addObject("factorList", factorMethods);
+   modelAndView.addObject("authenticators", response.getAuthenticators());
+   modelAndView.addObject("title", title);
+   return modelAndView;
+}
+```
+
+### Step 4: Enroll required authenticator
+
+Pass the user-selected authenticator (in this case, the password authenticator) to the `IDXAuthenticationWrapper.selectAuthenticator` method.
+
+```java
+authenticationResponse = idxAuthenticationWrapper.selectAuthenticator(proceedContext, authenticator);
+```
+
+This request returns an `AuthenticationResponse` object with `AuthenticationStatus=AWAITING_AUTHENTICATOR_VERIFICATION`. You need to build a form for the user to enter their password in this authenticator verification step.
+
+<div class="common-image-format">
+
+![Displays the Java SDK password authenticator verification form](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-self-serv-screen-confirm-password-java.png)
+
+</div>
+
+### Step 5: Verify authenticator and display additional factors
+
+After the user enters their new password, call the `IDXAuthenticationWrapper.verifyAuthenticator` method with the user's password.
+
+```java
+VerifyAuthenticatorOptions verifyAuthenticatorOptions = new VerifyAuthenticatorOptions(newPassword);
+AuthenticationResponse authenticationResponse = idxAuthenticationWrapper.verifyAuthenticator(proceedContext, verifyAuthenticatorOptions);
+```
+
+The request returns an `AuthenticationResponse` object with `AuthenticationStatus=AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION` and an `Authenticators` list containing the email and phone factors. Reuse the authenticator enrollment form from [Step 3: Display enrollment factors](#step-3-display-enrollment-factors) to display the list of authenticators to the user.
+
+<div class="common-image-format">
+
+![Displays the Java SDK Email and Phone authenticator enrollment form](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-self-serv-screen-auth-list-email-phone-java.png)
+
+</div>
+
+### Step 6: User selects email authenticator
+
+In this use case, the user selects the **Email** factor as the authenticator to verify. Pass this user-selected authenticator to the `IDXAuthenticationWrapper.selectAuthenticator` method.
+
+```java
+authenticationResponse = idxAuthenticationWrapper.selectAuthenticator(proceedContext, authenticator);
+```
+
+If this request is successful, a code is sent to the user's email and `AuthenticationStatus=AWAITING_AUTHENTICATOR_VERIFICATION` is returned. You need to build a form to capture the code for this verification step.
```

Comment:
reuse code from step 3? Or no?

Reply:
@vanngo-okta ?

Reply:
No, I think step 3 is the enrollment selection form, not the code verify form.